qcodes/dataset/plotting.py

```diff
@@ -37,6 +38,44 @@
 SUBPLOTS_KWARGS = SUBPLOTS_OWN_KWARGS.union(FIGURE_KWARGS)
 
 
+@contextmanager
+def appropriate_kwargs(plottype: str,
+                       ax: matplotlib.axes.Axes,
+                       colorbar: Optional[matplotlib.colorbar.Colorbar] = None,
+                       **kwargs):
+    """
+    NB: Only to be used inside :meth"`plot_by_id`.
+
+    Context manager to temporarily mutate the plotting kwargs to be appropriate
+    for a specific plottype. This is helpful since :meth:`plot_by_id` may have
+    to generate different kinds of plots (e.g. heatmaps and line plots) and
+    the user may want to specify kwargs only relevant to some of them
+    (e.g. 'cmap', that line plots cannot consume). Those kwargs should then not
+    be passed to all plots, which is what this contextmanager handles.
+
+    Args:
+        plottype: The plot type for which the kwargs should be adjusted
+        ax: The ax that is to be plotted on
+        colorbar: The colorbar that is to be used (if any)
+    """
+
+    def linehandler(**kwargs):
+        kwargs.pop('cmap', None)
+        return kwargs
+
+    def heatmaphandler(**kwargs):
+        if colorbar is None and 'cmap' not in kwargs:
+            kwargs['cmap'] = qc.config.plotting.default_color_map
+        return kwargs
+
+    switch = {'line': linehandler,
```

Comment:
Could we call the variable something else? `plot_handler_mapping` or some such thing. I know switch is not a keyword in python but for non python devs this may be a bit confusing